hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -151,4 +151,60 @@ public boolean apply(String taskId) {
       }
     }).size();
   }
+
+  private class NameNodeState {
+    private String name;
+    private TaskStatus status;
+    private Boolean initialized;
+
+    public NameNodeState(String name) {
+      this.name = name;
+      clear();
+    }
+
+    public Boolean hasStatus() {
+      return status != null;
+    }
+
+    public Boolean initialized() {
+      return status != null && initialized;
+    }
+
+    public void set(TaskStatus status, Boolean initialized) {
+      this.status = status;
+      this.initialized = initialized;
+      log.info(String.format("Set Node '%s' state to status: '%s', initialized: '%s'", name, status, initialized));
+    }
+
+    public void clear() {
+      set(null, false);
+    }
+
+    public void clear(TaskID taskId) {
+      if (status == null) {
+        return;
+      } else {
```

Comment:
To have a lot less lines of code just call if(hasStatus) No need for an if and an else